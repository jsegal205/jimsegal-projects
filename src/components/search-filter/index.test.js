import React from "react";
import { render, cleanup } from "@testing-library/react";
import SearchFilter from "./index";

afterEach(cleanup);

describe("Loading Component", () => {
  it("renders default state correctly", () => {
    const { asFragment } = render(<SearchFilter />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe("when `searchValue` is passed", () => {
    it("renders reset button", () => {
      const { asFragment } = render(
        <SearchFilter searchValue="test" handleSearchChange={() => {}} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
