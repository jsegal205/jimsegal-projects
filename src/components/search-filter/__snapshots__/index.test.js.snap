// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component renders default state correctly 1`] = `
<DocumentFragment>
  <section
    class="search-filter-container"
  >
    <label
      for="search-filter"
    >
      Search
    </label>
    <input
      class="search-filter-input"
      data-testid="search-filter"
      id="search-filter"
      type="text"
      value=""
    />
  </section>
</DocumentFragment>
`;

exports[`Loading Component when \`searchValue\` is passed renders reset button 1`] = `
<DocumentFragment>
  <section
    class="search-filter-container"
  >
    <label
      for="search-filter"
    >
      Search
    </label>
    <input
      class="search-filter-input"
      data-testid="search-filter"
      id="search-filter"
      type="text"
      value="test"
    />
    <button
      class="search-filter-reset"
      data-testid="search-filter-reset"
    >
      Reset
    </button>
  </section>
</DocumentFragment>
`;
